src/main/java/learnositysdk/request/Init.java

```diff
@@ -424,9 +428,34 @@ private void setRequestPacket(Object requestPacket) throws Exception
     private void updateRequestString() {
         this.requestString = this.requestPacket.toString();
 
-        // Undo escaped forward slashes by JSONObject.toString to match PHP sdk
-        // https://github.com/Learnosity/learnosity-sdk-php/blob/42f2ffade51831966e0f92f25fa06e50c1dbecce/src/LearnositySdk/Utils/Json.php#L54
-        this.requestString = this.requestString.replace("\\/", "/");
+        this.requestString = fixJson(this.requestString);
+    }
+
+    /**
+     * Ensures encoded JSON is correctly formatted for the Learnosity APIs
+     *
+     * PHP SDK: https://github.com/Learnosity/learnosity-sdk-php/blob/42f2ffade51831966e0f92f25fa06e50c1dbecce/src/LearnositySdk/Utils/Json.php#L54
+     */
+    private String fixJson(String string) {
+        final Pattern pattern = Pattern.compile("\\\\u([0-9A-Fa-f]{4})");
+        StringBuilder result = new StringBuilder();
+        final Matcher matcher = pattern.matcher(string);
+        int lastMatch = 0;
+
+        while(matcher.find())
+        {
+            final MatchResult matchResult = matcher.toMatchResult();
+            String replacement = Character.toString((char)Integer.parseInt(matchResult.group(1), 16));
```

Comment:
Do we know if this will always be based on UTF8 encoding and not possibly some other encoding where it could be cast to the wrong character?

Reply:
So according to the Java docs...
https://docs.oracle.com/javase/tutorial/java/nutsandbolts/datatypes.html
`char: The char data type is a single 16-bit Unicode character. It has a minimum value of '\u0000' (or 0) and a maximum value of '\uffff' (or 65,535 inclusive).`
Our regex matches that pattern, so for example `\uffff` would be matched, converted to 65535, which would then be converted into a unicode character and added to the string.

Reply:
Awesome